src/index.test.js

```diff
@@ -0,0 +1,66 @@
+// import React from 'react';
+import ReactDOM from 'react-dom';
+import Adapter from 'enzyme-adapter-react-16';
+import * as createjs from 'createjs-module';
+import { configure } from 'enzyme';
+import AccessibilityModule from './index';
+
+configure({ adapter: new Adapter() });
+
+describe('AccessibilityModule', () => {
+  describe('setupStage', () => {
+    let canvasEl;
+    let stage;
+    let container;
+    let parentElement;
+    let renderSpy;
+
+    beforeEach(() => {
+      canvasEl = document.createElement('canvas');
+      stage = new createjs.Stage(canvasEl);
+      container = new createjs.Container();
+      parentElement = document.createElement('div');
+      parentElement.id = 'parent-element';
+      renderSpy = jest.spyOn(ReactDOM, 'render');
+    });
+
+    it('calls ReactDOM.render and renders inside parentElement', () => {
+      expect(parentElement.childElementCount).toEqual(0);
+
+      AccessibilityModule.register({
+        displayObject: container,
+        role: 'main',
+      });
+      AccessibilityModule.setupStage(stage, parentElement, () => {
+        stage.accessibilityTranslator.root = container;
+      });
+
+      expect(renderSpy).toHaveBeenCalledTimes(1);
+      expect(parentElement.childElementCount).toEqual(1);
+    });
+
+    it('find parentElement by id if passed a string', () => {
+      const getElementByIdSpy = jest.spyOn(document, 'getElementById').mockReturnValue(parentElement);
+
+      AccessibilityModule.register({
+        displayObject: container,
+        role: 'main',
+      });
+      AccessibilityModule.setupStage(stage, parentElement.id);
+
+      expect(getElementByIdSpy).toBeCalledWith(parentElement.id);
+    });
```

Comment:
Just trying to understand what's going on here, is this accurate:

- 42-43: getElementByIdSpy creates a mock function that tracks `document.getElementById` calls and has a return value of `parentElement`, the div with id=parent-element that was set in `beforeEach`
- 45-48: give container role=main
- 49: in `setupStage`, it uses `document.getElementById` to find the `parentElement` object, which is tracked by the `spyOn` mock function
- 51: compares `getElementByIdSpy`'s parameter with `parentElement.id`, which should be equal because we gave `parentElement.id` as a param for `setupStage`, which calls `document.getElementById` with `parentElement.id`

Is `getElementByIdSpy`'s mock return value used for anything here if we're just checking its parameter? 

Is this also meant to check that the `parentElement` that `setupStage` renders `moduleNode` into is the same as the `parentElement` defined for the tests? If so, would we need to add something like:
`expect(getElementByIdSpy.mock.results[0].value).toBe(parentElement)` (might not be accurate, just a guess from a quick peek into the docs) to check the actual return value of the call rather than using our mock value?

Let me know if none of that makes sense, just trying to learn how Jest works and I'm a bit confused about how spyOn works with mockReturnValue 😅

Reply:
> Just trying to understand what's going on here, is this accurate:
> * 45-48: give container role=main

register `displayObject<createjs.Container>` to CAM with `role` of `main`

> Is `getElementByIdSpy`'s mock return value used for anything here if we're just checking its parameter?

not sure what you're asking here

> Is this also meant to check that the `parentElement` that `setupStage` renders `moduleNode` into is the same as the `parentElement` defined for the tests?

yes (although not sure where you're getting `moduleNode`)

your other bullet pointed assumptions are correct

Reply:
Thanks for the info! I think I'm just not sure how mock functions work yet. My understanding was that the       `.mockReturnValue(parentElement)` in line 43 makes `getElementByIdSpy` always return `parentElement` as its mock return value. I was wondering if that mock return value is actually used in this test. It looks like we only check the parameter passed to `document.getElementById` using `toBeCalledWith`. That's why I asked if it's also meant to check that the actual and expected `parentElement`s are the same, because I don't think that's what we're checking right now. I mention `moduleNode` because it's what gets rendered into `parentElement` in `setupStage`: https://github.com/CurriculumAssociates/createjs-accessibility/blob/1895411214ff4420596600336deb73b376bf5825/src/index.js#L67

I might just be misunderstanding what the test is doing/how mock functions work because I'm new to Jest, so please let me know if that's the case 😅

Reply:
It's necessary to make sure the ReactDOM.render doesn't subsequently break/cause the test to fail, but ensuring that there's an accurate DOM note that's returned by `getElementById`.

Reply:
gotcha - I think I understand your confusion now

adding to what @mmoss just said, if you pull this branch and inspect what's returning, you'll see that the mock is necessary because without it, when `parentElement` is reassigned in line 76 it would otherwise be `null`

this is the downside to how the original code was written in terms of testability

Reply:
ahhh I see, that clears it up! tysm for the explanation!